### ISSUE TITLE
This video contains content from ELLO. It is restricted from playback on certain sites.

### ISSUE THREAD

**Issue by artemmelnyk** [youtube/youtube-ios-player-helper#52]
I have this message "This video contains content from ELLO. It is restricted from playback on certain sites.". But in another programms videos from ELLO is playing. What can I do?

**Comment by JV17** [comment]
One option could be to just skip the video if you get an error message or load new one. You can do this by intercepting the YTPlayerView requests.. [Example here!](http://stackoverflow.com/questions/27820123/bug-in-youtube-ios-player-helper-library/27840798#27840798)

**Comment by Kowaio** [comment]
Hi @JV17,

About this matter, I was wondering how you could detect that the content is forbidden in order to display a custom message inside the app when this is the case.

Indeed, I don't receive a callback from this delegate "- (void)playerView:(YTPlayerView *)playerView receivedError:(YTPlayerError)error", currently.

Thanks a lot for your help.

Best,

**Comment by JV17** [comment]
@Kowaio I will just check what it will get returned within the error or as mentioned above  and then just trigger a function that will display this message everytime you get that response.

**Comment by JALsnipe** [comment]
Your video is playing back fine for me using the latest `YTPlayerView` classes.

```obj-c
- (void)viewDidLoad {
    [super viewDidLoad];

   NSDictionary *playerVars = @{
                                 @"playsinline" : @1,
                                 @"showinfo" : @0,
                                 @"rel" : @0,
                                 @"modestbranding" : @1,
                                 @"controls" : @1
                                 };
    
    [self.playerView loadWithVideoId:@"ASM7f1E-dy8" playerVars:playerVars];
    
}
```


Logging:
```
2015-01-28 15:37:55.081 TestApp[2520:348788] YTPlayerView : playerViewDidBecomeReady
2015-01-28 15:37:55.082 TestApp[2520:348788] quality levels available:- (
    7
)
2015-01-28 15:37:55.182 TestApp[2520:348788] YTPlayerView : didChangeToState
2015-01-28 15:37:55.622 TestApp[2520:348788] YTPlayerView : didChangeToQuality
2015-01-28 15:37:55.623 TestApp[2520:348788] kYTPlaybackQualityMedium
2015-01-28 15:38:00.113 TestApp[2520:348788] YTPlayerView : didChangeToQuality
2015-01-28 15:38:00.115 TestApp[2520:348788] kYTPlaybackQualityMedium
2015-01-28 15:38:00.135 TestApp[2520:348788] YTPlayerView : didChangeToState
2015-01-28 15:38:00.136 TestApp[2520:348788] Started playback
```

**Comment by Kowaio** [comment]
Hello @JALsnipe,

Mine was this one: JrGBmjtOKlM

However, I think that it cannot be played on mobile devices due to some restriction of the author. I might be wrong though.

@JV17: I succeeded to detect it on the didChangeToState callback by checking the "kYTPlayerStateUnstarted". Thanks again for your help.

Best,

**Comment by JALsnipe** [comment]
Thanks for the new URL.  It doesn't appear that I get any error callback when trying to load that video:

```
2015-01-28 15:50:28.017 TestApp[2529:350224] YTPlayerView : playerViewDidBecomeReady
2015-01-28 15:50:28.018 TestApp[2529:350224] quality levels available:- (
    7
)
2015-01-28 15:50:28.096 TestApp[2529:350224] YTPlayerView : didChangeToState
2015-01-28 15:50:28.573 TestApp[2529:350224] YTPlayerView : didChangeToState
2015-01-28 15:50:28.573 TestApp[2529:350224] Player in Unstarted State
```

Maybe one of the other experts can chime in, but it looks like the class tries to play the video, something in the iFrame is blocking it, and the player returns to the unstarted state.

**Comment by Kowaio** [comment]
@JALsnipe: Yes indeed. I didn't received any error callback as well. Since I needed to detect it either way, the "Unstarted" state kind of does the trick in my case. For now...

But yeah, I think that this is coming from YT API and can't be overpassed.

**Comment by thidayatullah** [comment]
The MAIN Problem is the video can be played on Android!!

So my company is developing an app for iOS and Android, basically it's an app to play curated video from Youtube.  
And now I get this problem when certain video is playable on Android app but not in iOS app.  If it is caused by restriction of the video how is it possible to be played on Android?? 

Here is my video "gKHQ715IMyc"

**Comment by atulvuliv** [comment]
Add `player vars` while initializing youtube sdk:

    NSDictionary *playerVars = @{
                                 @"origin" : @"http://www.youtube.com",
                                 };
    [self.playerView loadWithVideoId:@"KOvoD1upTxM" playerVars:playerVars];


Enjoy!

**Comment by swl367** [comment]
@atulvuliv thanks, that did it for me.

**Comment by mayuritalaviya-nv** [comment]
@atulvuliv 
Thanks, your replay save my time.

**Comment by ibhavin** [comment]
@atulvuliv  that doesn't work for me :(

for this videoId: 9VoHLolUaPc

**Comment by umairali332** [comment]
Hey guys!

Any update on this issue why some videos are not playing in this video player whereas same videos being played using Android player. 

for me this video is not being played. 6fvYcRdhuBE
